pandas/core/groupby/groupby.py

```diff
@@ -336,13 +347,13 @@ def _group_selection_context(groupby):
     groupby._reset_group_selection()
 
 
-class _GroupBy(PandasObject, SelectionMixin):
+class _GroupBy(PandasObject, SelectionMixin, Generic[FrameOrSeries]):
```

Comment:
Hmm not sure I would consider _GroupBy a generic class - is this required as part of this change or just experimenting?